test/distributions/test_distributions.py

```diff
@@ -4617,8 +4631,15 @@ def setUp(self):
                 scipy.stats.weibull_min(c=positive_var2[0], scale=positive_var[0])
             ),
             (
-                Wishart(20 + positive_var[0], cov_tensor),  # scipy var for Wishart only supports scalars
-                scipy.stats.wishart(20 + positive_var[0].item(), cov_tensor),
+                # scipy var for Wishart only supports scalars
+                Wishart(
+                    (20 if version.parse(scipy.__version__) < version.parse("1.7.0") else 19) + positive_var[0],
```

Comment:
Why does the version number affect df - what happens if we just use 20?

Reply:
I searched their development history and found that scipy allowed `ndim - 1 < df < ndim` only after `ver 1.7.0` with no additional sampling code modifications. Since it seems critical to test numerical stability in the region, I forced tests to consider `ndim - 1 < df < ndim` as long as we are available.

I do not understand much of the CI of PyTorch github and I observed that scipy tests are always throw error asserting `df > ndim`.
Hope I can find a way to apply custom version for dependancies in CI tests.

Reply:
Thanks for explaining.